packages/flutter/lib/src/widgets/basic.dart

```diff
@@ -5890,47 +5894,134 @@ class MouseRegion extends SingleChildRenderObjectWidget {
   ///
-  ///  * This widget, which used to contain a pointer, has moved away.
-  ///  * A pointer that used to be within this widget has been removed.
-  ///  * A pointer that used to be within this widget has moved away.
+  ///  * This widget, which is being hovered by a pointer, has moved away.
+  ///  * A pointer that is hovering this widget has been removed.
+  ///  * A pointer that is hovering this widget has moved away.
   ///
   /// And is __not__ triggered by the following case,
   ///
-  ///  * This widget, which used to contain a pointer, has disappeared.
+  ///  * This widget, which is being hovered by a pointer, has disappeared.
+  ///
+  /// This means that a [MouseRegion.onExit] might not be matched by a
+  /// [MouseRegion.onEnter].
   ///
-  /// The last case is the only case when [onExit] does not match an earlier
-  /// [onEnter].
-  /// {@macro flutter.mouseTracker.onExit}
+  /// This restriction aims to prevent a common misuse: if [setState] is called
+  /// during [MouseRegion.onExit] without checking whether the widget is still
+  /// mounted, an exception will occur. This is because the callback is
+  /// triggered during the post-frame phase, at which point the widget has been
+  /// unmounted. Since [setState] is exclusive to widgets, the restriction is
+  /// specific to [MouseRegion], and does not apply to its lower-level
+  /// counterparts, [RenderMouseRegion] and [MouseTrackerAnnotation].
+  ///
+  /// There are a few ways to mitigate this restriction:
+  ///
+  ///  * If the hover state is completely contained within a widget that
+  ///    unconditionally creates this [MouseRegion], then this will not be a
+  ///    concern, since after the [MouseRegion] is unmounted the state is no
+  ///    longer used.
+  ///  * Otherwise, the outer widget very likely has access to the variable that
+  ///    controls whether this [MouseRegion] is present. If so, call [onExit] at
+  ///    the event that turns the condition from true to false.
+  ///  * In cases where the solutions above won't work, you can always
+  ///    override [State.dispose] and call [onExit], or create your own widget
+  ///    using [RenderMouseRegion].
+  ///
+  /// {@tool snippet}
+  /// The following example shows a stateful widget where the hover state is
+  /// completely contained within a widget that unconditionally creates a
+  /// `MouseRegion`. In this case, you can ignore this restriction.
+  ///
+  /// ```dart
+  /// class MyButton extends StatefulWidget {
+  ///   @override
+  ///   State<StatefulWidget> createState() => _MyButtonState();
+  /// }
+  ///
+  /// class _MyButtonState extends State<MyButton> {
+  ///   bool hovered = false;
+  ///
+  ///   @override
+  ///   Widget build(BuildContext context) {
+  ///     return MouseRegion(
+  ///       onEnter: (_) { setState(() { hovered = true; }); }
+  ///       onExit: (_) { setState(() { hovered = false; }); }
+  ///     );
+  ///   }
+  /// }
+  /// ```
+  ///
```

Comment:
You should end this with `{@end-tool}` and start a new one for the next example.